source/core_mqtt.c

```diff
@@ -1386,7 +1408,7 @@ static MQTTStatus_t sendPublish( MQTTContext_t * pContext,
                             pContext->networkBuffer.pBuffer,
                             headerSize );
 
-    if( bytesSent < 0 )
+    if( bytesSent < ( int32_t ) headerSize )
```

Comment:
Are you going to make the same change in the other places where `sendPacket` is called, or will that be a separate PR?

Reply:
Separate PR to avoid scope screep in this PR.